### PR TITLE
Include target traits in spell damage rolls and allow spell REs to use "{item|_id}-damage" selector

### PR DESCRIPTION
**Pull request by simonward**

### CODE REVIEW

File: src/module/item/spell/document.ts [hunk 1]
```diff
@@ -261,8 +261,15 @@ class SpellPF2e extends ItemPF2e {
 
         const { actor, ability } = this;
         const domains = ["damage", "spell-damage"];
+        const targetToken =
+            Array.from(game.user.targets).find((t) => t.actor?.isOfType("creature", "hazard", "vehicle")) ?? null;
         const options = new Set<string>(
```
Comment (stwlam) [inline src/module/item/spell/document.ts:266]: Remove the `<string>` generic parameter

File: src/module/item/spell/document.ts [hunk 2]
```diff
@@ -260,9 +260,16 @@ class SpellPF2e extends ItemPF2e {
...
+        const targetToken =
+            Array.from(game.user.targets).find((t) => t.actor?.isOfType("creature", "hazard", "vehicle")) ?? null;
+        const options = new Set(
+            [
+                actor?.getRollOptions(domains) ?? [],
+                this.getRollOptions("item"),
+                [...this.traits],
+                targetToken?.actor?.getSelfRollOptions("target") ?? [],
+            ].flat()
         );
```
Comment (CarlosFdez) [inline src/module/item/spell/document.ts:273]:
```suggestion
        const options = new Set([
            ...(actor?.getRollOptions(domains) ?? []),
            ...(targetToken?.actor?.getSelfRollOptions("target") ?? []),
            ...this.getRollOptions("item"),
            ...this.traits,
        ]);
```

**Comment by CarlosFdez** [comment]
It may be best to make target a parameter for getDamage(). As written right now, if there is a target, then even the buttons in the character sheet will show target damage when opened until the next time its re-rendered. We only want a target to exist when we are rolling damage.

**Comment by simonward** [comment]
> It may be best to make target a parameter for getDamage(). As written right now, if there is a target, then even the buttons in the character sheet will show target damage when opened until the next time its re-rendered. We only want a target to exist when we are rolling damage.

Good call.

**Comment by CarlosFdez** [comment]
Might be best to add target as an option to SpellDamageOptions (that allows null) and then do `{ target, ...eventToRollParams(event) }`. target is optional, so it might as well go in options.

**Comment by simonward** [comment]
> Might be best to add target as an option to SpellDamageOptions (that allows null) and then do `{ target, ...eventToRollParams(event) }`. target is optional, so it might as well go in options.

Sounds good.

**Pull request merged**